docs/BreakingChanges/042 Enumerable.Empty returns cached instance.md

```diff
@@ -0,0 +1,24 @@
+## 42: Enumerable.Empty<TResult> method
```

Comment:
{
  "Id": "42",
  "Title": "Enumerable.Empty<TResult> method",
  "Details": "Beginning in .NET 4.5, `Enumerable.Empty` always returns a cached internal instance `IEnumerable<T>`.\n\nPreviously, `Enumerable.Empty` would cache an empty `IEnumerable<T>` at the time the API was called, meaning that in some conditions in which `Enumerable.Empty` was called rapidly and concurrently, different instances of the type could be returned for different calls to the API.",
  "Suggestion": "Because the previous behavior was non-deterministic, code is unlikely to depend on it. However, in the unlikely case that empty enumerables are being compared and expected to sometimes be unequal, explicit empty arrays should be created (`new T[0]`) instead of using `Enumerable.Empty`.",
  "Link": "https://msdn.microsoft.com/en-us/library/hh367887(v=vs.110).aspx#sql",
  "Markdown": null,
  "ApplicableApis": [
    "M:System.Linq.Enumerable.Empty`1"
  ],
  "Related": null,
  "VersionBroken": "4.5",
  "VersionFixed": null,
  "IsBuildTime": false,
  "IsQuirked": false,
  "IsSourceAnalyzerAvailable": true,
  "BugLink": null,
  "Notes": null,
  "IsRetargeting": false,
  "ImpactScope": 3
}

**Parsed APIs:**
- [ ] M:System.Linq.Enumerable.Empty`1